controller/engine_controller.go

```diff
@@ -1710,6 +1712,11 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 			// it is past the Backoff period, we'll try to mark the Replica as Failed and increase the Backoff period
 			// for the next failure.
 			if !ec.backoff.IsInBackOffSinceUpdate(e.Name, time.Now()) {
+				if err := updateReplicaRebuildFailedRebuildStatus(log, ec.ds, replica, replicaRebuildErrMsg); err != nil {
```

Comment:
I am thinking if Longhorn should record the rebuilding error immediately after the failure